# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsNSSComponent.h
# Commit: 20f3d7a5481d
# Full Hash: 20f3d7a5481db59c20e27a1603a58a40fdad9c93
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-02-10 21:33:47
# Regressor Bug: 1607194
# File Overlap Count: 1
# Description:
#   Bug 1607194 - P1: NSS initialization without nsNSSComponent for  the socket process r=keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D58774
# ==============================================================================

diff -r 79b7b94df8b6 -r 20f3d7a5481d security/manager/ssl/nsNSSComponent.h
--- a/security/manager/ssl/nsNSSComponent.h	Mon Feb 10 15:53:23 2020 +0000
+++ b/security/manager/ssl/nsNSSComponent.h	Mon Feb 10 17:07:26 2020 +0000
@@ -49,6 +49,9 @@
   }
 
 extern bool EnsureNSSInitializedChromeOrContent();
+extern bool HandleTLSPrefChange(const nsCString& aPref);
+extern void SetValidationOptionsCommon();
+extern void NSSShutdownForSocketProcess();
 
 // Implementation of the PSM component interface.
 class nsNSSComponent final : public nsINSSComponent, public nsIObserver {
@@ -74,6 +77,8 @@
                                   uint32_t minFromPrefs, uint32_t maxFromPrefs,
                                   SSLVersionRange defaults);
 
+  static nsresult SetEnabledTLSVersions();
+
  protected:
   virtual ~nsNSSComponent();
 
@@ -84,7 +89,6 @@
   void setValidationOptions(bool isInitialSetting,
                             const mozilla::MutexAutoLock& proofOfLock);
   void UpdateCertVerifierWithEnterpriseRoots();
-  nsresult setEnabledTLSVersions();
   nsresult RegisterObservers();
 
   void MaybeImportEnterpriseRoots();